packages/internal/src/generate/graphqlCodeGen.ts

```diff
@@ -208,8 +208,25 @@ function getCodegenOptions(
   config: CodegenTypes.PluginConfig,
   extraPlugins: CombinedPluginConfig[]
 ) {
+
+  // Because we want to generate enums as types we configure the typescript plugin
+  // to treat enums as types using the `enumsAsTypes` option set to true by default.
+  //
+  // However, since we allow the project to override the codegen configuration,
+  // (see: https://redwoodjs.com/docs/project-configuration-dev-test-build#graphql-codegen)
+  // we need to check if the project's custom codegen.yml has set this option.
+  //
+  // If not, set it to true otherwise set it to the config value, so it can be set to false
+  // so codegen generates an enum rather than a type.
+  //
+  // Note: If any additional typescript plugin configuration settings
+  // (see: https://www.graphql-code-generator.com/plugins/typescript/typescript)
+  // are defaulted, then they will need to be added here as well.
+
+  const enumsAsTypes = config.enumsAsTypes ?? true
+
   const plugins = [
-    { typescript: { enumsAsTypes: true } },
+    { typescript: { enumsAsTypes } },
```

Comment:
So something I got reminded of recently - with codegen plugins, the config does _not_ necessarily need to be at the plugin level. 

If we just moved `enumAsTypes` to pluginConfig https://github.com/dac09/redwood/blob/b6e58273cc3f920d8bd7583dfa5898467bf10eec/packages/internal/src/generate/graphqlCodeGen.ts#L196 instead of inside the extraPlugins, I think we'd get this functionality automatically. 

Note that the default for this is false.

Reply:
Also..... I don't think you can use enums from dts files...  which is the reason (I think) we didn't produce enums. There was a

To verify this:
- write some code that uses an enum in the api side
- run yarn rw build and yarn rw serve
- I think.... it'll crash. 

Here's the old me talking about this: https://github.com/redwoodjs/redwood/issues/2788#issuecomment-861450759

But worth verifying incase things have changed

Reply:
>  don't think you can use enums from dts files

I read that as well ...

> Generates enum as TypeScript type instead of enum. Useful if you wish to generate .d.ts declaration file instead of .ts


But, I just scaffolded a model called Project


```ts
export const schema = gql`
  type Project {
    id: Int!
    name: String!
    status: NameStatus!
  }

  enum NameStatus {
    NAME_REQUESTED
    NAME_APPROVED
    NAME_DENIED
    NAME_CHANGING
  }

  type Query {
    projects: [Project!]! @skipAuth
    project(id: Int!): Project @skipAuth
  }

  input CreateProjectInput {
    name: String!
    status: NameStatus!
  }

  input UpdateProjectInput {
    name: String
    status: NameStatus
  }

  type Mutation {
    createProject(input: CreateProjectInput!): Project! @skipAuth
    updateProject(id: Int!, input: UpdateProjectInput!): Project! @skipAuth
    deleteProject(id: Int!): Project! @skipAuth
  }
`
```

where inside the d.ts files it has 


```
export enum NameStatus {
  NAME_APPROVED = 'NAME_APPROVED',
  NAME_CHANGING = 'NAME_CHANGING',
  NAME_DENIED = 'NAME_DENIED',
  NAME_REQUESTED = 'NAME_REQUESTED'
}
```

And scaffolding all worked!


```api | 16:48:21 🐛 Processing GraphQL Parameters
api | 16:48:21 🐛 Execution start
api | 16:48:21 🐛 Received GraphQL operation:
api | 16:48:21 🐛 undefined 
api | 🏷  CreateProjectMutation 
api | 🔭 Query
api | "mutation CreateProjectMutation($input: CreateProjectInput!) {\n  createProject(input: $input) {\n    id\n    __typename\n  }\n}"
api | 16:48:21 🐛 graphql-server GraphQL execution started: CreateProjectMutation
api | 16:48:21 🐛 Execution end
api | 16:48:21 🌲 request completed 350ms
api | 16:48:21 🐛 undefined
api | 16:48:21 🐛 graphql-server GraphQL execution completed: CreateProjectMutation
api | 16:48:21 🌲 incoming request POST xxx /graphql
api | 16:48:21 🐛 Parsing request to extract GraphQL parameters
api | 16:48:21 🐛 Processing GraphQL Parameters
api | 16:48:21 🐛 Execution start
api | 16:48:21 🐛 Received GraphQL operation:
api | 16:48:21 🐛 undefined 
api | 🏷  FindProjects 
api | 🔭 Query
api | "query FindProjects {\n  projects {\n    id\n    name\n    status\n    __typename\n  }\n}"
api | 16:48:21 🐛 graphql-server GraphQL execution started: FindProjects
api | 16:48:22 🐛 Execution end
api | 16:48:22 🌲 request completed 9ms
api | 16:48:22 🐛 undefined
api | 16:48:22 🐛 graphql-server GraphQL execution completed: FindProjects
```

![image](https://user-images.githubusercontent.com/1051633/186520936-1e4f97e7-a865-4de4-94ca-3d4add6b8c27.png)


![image](https://user-images.githubusercontent.com/1051633/186521012-6efd6d87-730a-48ca-a017-508ecc8f3bb0.png)


![image](https://user-images.githubusercontent.com/1051633/186521090-2ce9e155-8645-4dcf-ba7b-abfc15e79ecb.png)


and then edited

![image](https://user-images.githubusercontent.com/1051633/186521138-f1372582-9ac9-477e-a7d9-254a42211158.png)


So, it does seem to work ok.


I also switched the flag to ...

```

export type NameStatus =
  | 'NAME_APPROVED'
  | 'NAME_CHANGING'
  | 'NAME_DENIED'
  | 'NAME_REQUESTED';

```

And also worked. :)

Reply:
On api side, in a service:


```ts
import { NameStatus } from '@prisma/client'
import type { QueryResolvers, MutationResolvers } from 'types/graphql'

import { db } from 'src/lib/db'
import { logger } from 'src/lib/logger'

export const projects: QueryResolvers['projects'] = () => {
  return db.project.findMany()
}

export const project: QueryResolvers['project'] = async ({ id }) => {
  const result = await db.project.findUnique({
    where: { id },
  })

  if (result.status === NameStatus.NAME_APPROVED) {
    logger.info(`project ${result.status}`)
  }

  return result
}
```

Appears to work ok with

```
// api/types/graphql.d.ts
export enum NameStatus {
  NAME_APPROVED = 'NAME_APPROVED',
  NAME_CHANGING = 'NAME_CHANGING',
  NAME_DENIED = 'NAME_DENIED',
  NAME_REQUESTED = 'NAME_REQUESTED'
}
```


```
 🏷  FindProjectById 
api | 🔭 Query
api | "query FindProjectById($id: Int!) {\n  project: project(id: $id) {\n    id\n    name\n    status\n    __typename\n  }\n}"
api | 11:40:11 🐛 graphql-server GraphQL execution started: FindProjectById 
api | 🔭 Query
api | {
api |   "id": 2
api | }
api | 11:40:11 🌲 project NAME_APPROVED
api | 11:40:11 🌲 request completed 7ms
api | 11:40:11 🐛 Execution end
api | 11:40:11 🐛 undefined
api | 11:40:11 🐛 graphql-server GraphQL execution completed: FindProjectById 
api | 🔭 Query
api | {
api |   "id": 2
api | } 
api | 📦 Result Data
api | {
api |   "project": {
api |     "id": 2,
api |     "name": "Project X",
api |     "status": "NAME_APPROVED",
api |     "__typename": "Project"
api |   }
api | }****
```

Reply:
And `yarn rw serve` works, too.

Reply:
Oh, that's from Prisma. Let me try importing the type, too, I imagine.